### PR TITLE
[Cookies] Remove CookieView-related files from the settings page

### PR DESCRIPTION
**Pull request by jagadeshjai**
- Remove CookieView-related files from the settings page as its support has been removed from upstream.
- This is an follow up for this commit: https://github.com/brave/brave-core/commit/e4486716a0dced3a1b79d3372b4bc0c6e009f391

Resolves https://github.com/brave/brave-browser/issues/39104
Resolves https://github.com/brave/brave-browser/issues/39033

Outcome:

[brave_cookie_settings_after_fix.webm](https://github.com/brave/brave-core/assets/45998266/27202173-c811-4e0b-bb9c-9023e67360a8)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by jagadeshjai** [comment]
@iefremov Could you please run the automation test and confirm that these changes work?

File: browser/resources/settings/brave_overrides/site_details.ts [hunk 1]
```diff
@@ -151,26 +145,6 @@ RegisterPolymerTemplateModifications({
     const usageSection = templateContent.querySelector('div#usage')
     if (!usageSection) {
       console.error(`[Brave Settings Overrides] Couldn't find usageSection item`)
```
Comment (emerick) [inline browser/resources/settings/brave_overrides/site_details.ts:147]: I think we can delete this code as well, as it's no longer used from what I can see.

**Comment by emerick** [comment]
This looks great!

I think it needs to be rebased, as we merged cr127 today. Please add a "Resolves https://github.com/brave/brave-browser/issues/39033" to the description, as it also fixes that issue.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by jagadeshjai** [comment]
> @jagadeshjai this looks great!
> 
> I think it needs to be rebased, as we merged cr127 today. Please add a "Resolves [brave/brave-browser#39033](https://github.com/brave/brave-browser/issues/39033)" to the description, as it also fixes that issue.

Done @emerick .

File: patches/chrome-browser-resources-settings-router.ts.patch [hunk 2]
```diff
@@ -6,7 +6,8 @@ index 7e03316d8975291611bb095bd3b87c5133f1e33b..6b604189c90f73f0bdffcfcf05edfc73
    IMPORT_DATA: Route;
    SIGN_OUT: Route;
    // </if>
-+  GET_STARTED: Route; SHIELDS: Route; SOCIAL_BLOCKING: Route; EXTENSIONS: Route; EXTENSIONS_V2: Route; BRAVE_SYNC: Route; BRAVE_SYNC_SETUP: Route; BRAVE_IPFS: Route; BRAVE_WALLET: Route; BRAVE_WEB3: Route; BRAVE_NEW_TAB: Route; THEMES: Route; BRAVE_SITE_SETTINGS_COOKIES_DETAILS: Route;
++  GET_STARTED: Route; SHIELDS: Route; SOCIAL_BLOCKING: Route; EXTENSIONS: Route; EXTENSIONS_V2: Route; BRAVE_SYNC: Route; BRAVE_SYNC_SETUP: Route; BRAVE_IPFS: Route; BRAVE_WALLET: Route; BRAVE_WEB3: Route; BRAVE_NEW_TAB: Route; THEMES: Route;
+==== BASE ====
```
Comment (emerick) [inline patches/chrome-browser-resources-settings-router.ts.patch:10]: This seems wrong. Is this line meant to be here?

Comment (jagadeshjai) [reply patches/chrome-browser-resources-settings-router.ts.patch:10]: Oops! I overlooked after the fixing the conflicts.

**Review by emerick — APPROVED** [review]

**Comment by jagadeshjai** [comment]
@emerick Could you please run CI for this and then merge?

**Comment by jagadeshjai** [comment]
@emerick I guess you need to create a duplicate PR for CI like this one https://github.com/brave/brave-core/pull/24236, since I don't have the permission.

**Comment by emerick** [comment]
@bsclifton Would you be able to find someone to help @jagadeshjai with CI on this while I'm on PTO? Thank you!

**Comment by bsclifton** [comment]
Sure - let me get CI rolling on this 😄

**Comment by bsclifton** [comment]
CI looks great - going to go ahead and merge. Nice work, @jagadeshjai!

**Comment by ShivanKaul** [comment]
This is great work @jagadeshjai :)

**Comment by ShivanKaul** [comment]
@bsclifton we should probably uplift this since it's broken in Release.

**Comment by bsclifton** [comment]
Super late to the game 😅 but compiled locally and verified this works great

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.69.72 Chromium: 127.0.6533.26 (Official Build) nightly (x86_64)
-- | --
Revision | 334f959c74f8e09df9e6f355d0ca2df9409b0303
OS | macOS Version 13.6.7 (Build 22G720)
```

**Pull request merged**